Administrator remove fix

JohnCox2211 opened the pull request:
Fixes issue #530 

Refactored code that deals with adding and removing admins in order to make it more robust for possible future modifications and prevents improper admin removals that are currently possible.

JohnCox2211 committed:
converted js to jquery for consistency

JohnCox2211 committed:
Split up the form in adminManagement.html into
three separate forms, one for each admin type,
to allow for proper removal of each type.

JohnCox2211 committed:
- removed leading spaces in form IDs
- updated modal IDs and button type
- refactored logic in adminManagement.js to be
more concise and reusable, fixing the issue in
the process.

JohnCox2211 committed:
optimzation for python adminManagement code,
will fully test when I get Tracy working

bakobagassas commented:
On your branch, when I add a name that is <3 letters and click enter and remove, instead of removing them, I receive a flash message saying they have been added as a labor admin. This is weird because I tried the same thing I did another time and it did not happened then I did it a third time and the problem happened that time.

bakobagassas commented:
Also when I click to add alex for example: (1) I put 3 characters, the name appeared (2) I removed one character, the name was still there (3) but I am still unable to actually click on his name although it is there on my screen.

JohnCox2211 commented:
> Also when I click to add alex for example: (1) I put 3 characters, the name appeared (2) I removed one character, the name was still there (3) but I am still unable to actually click on his name although it is there on my screen.

Upon testing, this bug that  you mentioned also appears in the organizingtest branch, so it is not new to our branch. We will, however, incorporate this change into our branch

JohnCox2211 commented:
> On your branch, when I add a name that is <3 letters and click enter and remove, instead of removing them, I receive a flash message saying they have been added as a labor admin. This is weird because I tried the same thing I did another time and it did not happened then I did it a third time and the problem happened that time.

I was unable to replicate this bug, could you show me next time we are both in the programmer space?

JohnCox2211 committed:
Merge branch 'organizingtest' of https://github.com/BCStudentSoftwareDevTeam/lsf into administrator-remove-fix

JohnCox2211 committed:
Merge branch 'organizingtest' of https://github.com/BCStudentSoftwareDevTeam/lsf into administrator-remove-fix

BrianRamsay reviewed (changes requested):
`getUser` is being imported from a file in `logic/`, but you've added it back in to the controller also. Was that a merge conflict issue?

JohnCox2211 committed:
You will no longer see a list of students after
removing characters from the search bgox if there
are less than 3 characters, fixing the issue of
being unable to select a student after removing
chars.

JohnCox2211 committed:
removed duplicate getUser function, also removed
unused add/remove admin function imports

JohnCox2211 commented:
> `getUser` is being imported from a file in `logic/`, but you've added it back in to the controller also. Was that a merge conflict issue?

Resolved this issue, deleted duplicate function in controller

BrianRamsay commented on app/controllers/admin_routes/adminManagement.py:
```diff
@@ -47,35 +47,39 @@ def adminSearch():
+    "addSAASAdmin":            {"type": "saas", "action": "add", "pretty": "SAAS"},
+    "removeSAASAdmin":         {"type": "saas", "action": "remove", "pretty": "SAAS"},
+    }
+    for key, meta in actionMap.items():
```
You don't need to loop through the dictionary if you only have one key. You can just select the values from the dictionary directly:
`meta = actionMap.get(request.form.get(key))`

JohnCox2211 replied:
Changed how I was handling the logic there and no longer need that loop there.

BrianRamsay commented on app/controllers/admin_routes/adminManagement.py:
```diff
@@ -47,35 +47,39 @@ def adminSearch():
+            user = getUser(key)
+            
+            # pick addAdmin or removeAdmin dynamically
+            (addAdmin if meta["action"] == "add" else removeAdmin)(user, meta["type"])
```
Let's keep this as a multi-line if/else

BrianRamsay commented on app/static/js/adminManagement.js:
```diff
@@ -43,56 +43,46 @@ function liveSearch(selectPickerID, e) {
+    $("#flash_container").html(
+      '<div class="alert alert-danger" role="alert" id="flasher">Please select a user.</div>'
+    );
+    $("#flasher").delay(3000).fadeOut();
```
Let's use our javascript flash message function

BrianRamsay commented on app/static/js/adminManagement.js:
```diff
@@ -43,56 +43,46 @@ function liveSearch(selectPickerID, e) {
+    remove2:{ select: "#removeSaasAdmin",    form: "#SAASAdminForm",    header: "SAAS Administrators",        action: "remove", noun: "SAAS Administrator" }
+  };
+
+  const cfg = config[button];
```
`config` and `cfg` are too close to each other. let's come up with more descriptive names, particularly for `cfg`.

JohnCox2211 replied:
Renamed config to formButtons and cfg to adminForm. These names seem to be better descriptors.

JohnCox2211 committed:
resolved most of brian's comments,
figuring out if i can remove the loop from
adminMapping in adminManagement.py

JohnCox2211 committed:
refactored code to remove loop in adminManagement
currently new version only works to add/remove
labor admins

JohnCox2211 committed:
all forms able to be submitted now with refactor

JohnCox2211 committed:
Figured out fin aid admins weren't being added
fixed that..

JohnCox2211 committed:
removed print.........

JohnCox2211 committed:
Merge branch 'development' of https://github.com/BCStudentSoftwareDevTeam/lsf into administrator-remove-fix

eunsungw committed:
Merge branch 'development' into administrator-remove-fix

BrianRamsay committed:
Reduce individual queries on page load

BrianRamsay commented:
I added a commit to change the number of queries on page load from 805 to 10.
